Detections/MultipleDataSources/UserImpersonateByRiskyUser.yaml

```diff
@@ -0,0 +1,69 @@
+name: High-Risk Cross-Cloud User impersonation 
+description: |
+  'This detection focuses on dentifying high-risk cross-cloud activities and sign-in anomalies that may indicate potential security threats. The query starts by analyzing Azure AD SigninLogs to pinpoint instances where specific applications, risk levels, and result types align. It then correlates this information with relevant AWS CloudTrail events to identify activities across Azure and AWS environments.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: AWS
+    dataTypes:
+      - AWSCloudTrail
+  - connectorId: AzureActiveDirectory
+    dataTypes:
+      - SigninLogs
+queryFrequency: 1d
+queryPeriod: 1d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - PrivilegeEscalation
+relevantTechniques:
+  - T1134
+  - T1078.002
+  - T1078.004
+query: |
+  // Retrieve Azure AD SigninLogs within the last day
+  SigninLogs 
+  | where TimeGenerated >= ago(1d)
+  // Filter for specific AppDisplayNames, ResultType, and Risk Levels
+  | where AppDisplayName in ("Azure Portal", "ADFS Trust", "Microsoft Azure PowerShell")
+      and ResultType == 0
+      and RiskLevelAggregated == "high"
+      and RiskLevelDuringSignIn == "high"
+  // Summarize AppDisplayNames by relevant attributes
+  | summarize make_set(AppDisplayName)
+      by
+      IPAddress,
+      signInTime=TimeGenerated,
+      UserPrincipalName,
+      RiskEventTypes,
+      RiskEventTypes_V2
+  // Inner join with AWS CloudTrail events
+  | join kind=inner (
+      AWSCloudTrail
+      | where TimeGenerated >= ago(1d)
+      | where isempty(ErrorMessage)
+      | where EventSource in ("iam.amazonaws.com", "identitystore.amazonaws.com", "workmail.amazonaws.com", "workdocs.amazonaws.com")
+      // List of AWS event names
+      | where EventName in~ ("CreateRole", "DeleteRole", "CreateUser", "CreateAccessKey", "DeleteAccessKey", "CreateGroup", "AddUserToGroup", "ChangePassword", "DeleteGroup", "DeleteUser", "RemoveUserFromGroup", "CreateVirtualMFADevice", "DeleteLoginProfile", "CreateOrganization", "SetDefaultMailDomain", "SetMailUserDetails", "CreateMailUser", "ResetPassword", "RegisterToWorkMail", "DisableMailUsers", "EnableMailUsers", "DeleteServiceSpecificCredential", "CreateServiceSpecificCredential", "UpdateAccountEmailAddress", "DeleteGroupPolicy", "UploadServerCertificate")  
+      // Summarize relevant attributes
+      | summarize make_set(RequestParameters), make_set(ResponseElements)
+          by
+          SourceIpAddress,
+          UserIdentityArn,
+          UserIdentityType,
+          EventName,
+          EventTime=TimeGenerated,
+          EventSource
+      )
+      on $left.IPAddress == $right.SourceIpAddress  
+  // Calculate time difference in hours between AWS event and Azure sign-in
+  | extend timedef = datetime_diff("hour", EventTime, signInTime)
+  // Filter for time differences within a certain range
+  | where timedef between (0 .. 8)
+entityMappings:

```

Comment:
What about the User entity here?

Reply:
added in customDetails

Reply:
I see that but why is it not also an Entity here? We use Entities for correlations and the custom details to add context. If we don't add user to entityMappings we can't use it for correlations.

Reply:
Actually, we are not able to add a user entity here as we do not have a use name separately in AWS, That is an issue we have raised that could be overcome by One Identity solution